python/ray/autoscaler/commands.py

```diff
@@ -415,8 +415,11 @@ def wrap_docker(command):
             attach_info = "Use `{}` to check on command status.".format(
                 attach_command)
             logger.info(attach_info)
-    finally:
+    except Exception as e:
         provider.cleanup()
+        raise e
+
+    provider.cleanup()
```

Comment:
Out of curiosity, what difference does this change make? It looks equivalent to me.

Reply:
Good catch, I forgot to revert this after debugging.